src/index.js

```diff
@@ -57,10 +60,19 @@ const createHoxy = () => {
     const cert = fs.readFileSync('./root-ca.crt.pem');
     opts.certAuthority = {key, cert};
   } catch (e) {
-    console.warn('Not proxying HTTPS, missing key or certificate:\n', e); // eslint-disable-line
+    data.proxyStatus = 'partial';
+    data.proxyMessage = 'no HTTPS';
   }
 
-  return hoxy.createServer(opts).listen(config.proxyPort);
+  const hoxyServer = hoxy.createServer(opts);
+  hoxyServer.on('error', (event) => {
+    data.proxyStatus = 'offline';
+    data.proxyMessage = event.code;
+    if (event.code === 'EADDRINUSE') {
+      // TODO make custom window
+    }
```

Comment:
I suddenly grok. In the `on` part, which is asynchronous. `render()` added, thanks :)